lib/Chat/Notifier.php

```diff
@@ -92,13 +93,49 @@ public function notifyMentionedUsers(Room $chat, IComment $comment): array {
 
 		$notification = $this->createNotification($chat, $comment, 'mention');
 		foreach ($mentionedUserIds as $mentionedUserId) {
+			if (in_array($mentionedUserId, $mentionedUsers, true)) {
+				continue;
+			}
+
 			if ($this->shouldUserBeNotified($mentionedUserId, $comment)) {
 				$notification->setUser($mentionedUserId);
 				$this->notificationManager->notify($notification);
+				$mentionedUsers[] = $mentionedUserId;
 			}
 		}
 
-		return $mentionedUserIds;
+		return $mentionedUsers;
+	}
+
+	/**
+	 * Notifies the author that wrote the comment which was replied to
+	 *
+	 * The comment must be a chat message comment. That is, its "objectId" must
+	 * be the room ID.
+	 *
+	 * Not every author mentioned in the message is notified, but only those that
+	 * are able to participate in the room.
```

Comment:
Maybe _The author of the message is notified only if he is still able to participate in the room_ instead?